internal/proto/volumegroup.proto

```diff
@@ -0,0 +1,111 @@
+  // This field is REQUIRED.
+  string name = 1;
+
+  // params passed to the plugin to create the volume group.
+  // This field is OPTIONAL.
+  map<string, string> parameters = 2;
+
+  // Secrets required by the driver to complete the
+  // create request.
+  string secret_name = 3;
+  string secret_namespace = 4;
+
+  // Specify volume_ids that will be added to the volume group.
+  // This field is OPTIONAL.
+  repeated string volume_ids = 5;
+}
+// CreateVolumeGroupResponse holds the information to send when
+// volumeGroup is successfully created.
+message CreateVolumeGroupResponse {
+  // Contains the identifier of the newly created volume group.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+}
+// DeleteVolumeGroupRequest holds the required information to
+// delete a volume group
+message DeleteVolumeGroupRequest {
+  // The ID of the volume group to be deleted.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Secrets required by the driver to complete the
+  // delete request.
+  string secret_name = 2;
+  string secret_namespace = 3;
+}
+// DeleteVolumeGroupResponse holds the information to send when
+// volumeGroup is successfully deleted.
+message DeleteVolumeGroupResponse {
+}
+// ModifyVolumeGroupMembershipRequest holds the required
+// information to modify a volume group
+message ModifyVolumeGroupMembershipRequest {
+  // The ID of the volume group to be modified.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Specify volume_ids that will be in the modified volume group.
+  // This list will be compared with the volume_ids in the existing
+  // group.
+  // New ones will be added and missing ones will be removed.
+  // If no volume_ids are provided, all existing volumes will
+  // be removed from the group.
+  // This field is OPTIONAL.
+  repeated string volume_ids = 2;
+
+  // Secrets required by the driver to complete the
+  // modification request.
+  string secret_name = 3;
+  string secret_namespace = 4;
+}
+// ModifyVolumeGroupMembershipResponse holds the information to
+// send when volumeGroup is successfully modified.
+message ModifyVolumeGroupMembershipResponse {
+  // Contains the identifier of the modified volume group.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+}
+// ControllerGetVolumeGroupRequest holds the required
+// information to get information on volume group
+message ControllerGetVolumeGroupRequest {
+  // The ID of the volume group to fetch current volume group
+  // information for.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Secrets required by the driver to complete the
+  // get request.
+  string secret_name = 2;
+  string secret_namespace = 3;
+}
+// ControllerGetVolumeGroupResponse holds the information to
+// send when volumeGroup information was successfully gathered.
+message ControllerGetVolumeGroupResponse {
+  // This field is REQUIRED
+  string volume_group_id = 1;
```

Comment:
What is the use of this? The same `volume_group_id` was passed in the request.

Reply:
We can remove this as well, since if the volume group is present we can check for an empty (succesfull) response and use the passed group_id.

Reply:
If this message is empty, why call the function at all?

I guess that at one point, you want to include all volumes that are part of the volume group in some VolumeGroupReplicationContent.Status kind of struct?

Reply:
Well, I am already adding all the volumes that are part of the volume group to the VGRContent.Status, by comparing the volume handles passed to the spec of the same.
Here: https://github.com/csi-addons/kubernetes-csi-addons/blob/8e8830de8f338f75e20dc8a1ec665e0440f78fb4/internal/controller/replication.storage/volumegroupreplicationcontent_controller.go#L160

Reply:
So, probably it is better to fetch the list using the `ControllerGetVolumeGroup` method instead of what I am doing right now?

Reply:
Ok, I guess that's not possible since we would be saving the volume names in the status and the info returned is just the group/volume ids.
So, I will just make the function empty. And, one can just use the function to validate that a volume group exists for the provided group_id.